opentelemetry-otlp/src/metric.rs

```diff
@@ -248,6 +258,36 @@ impl<RT, EB: Debug> Debug for OtlpMetricPipeline<RT, EB> {
     }
 }
 
+/// A temporality selector that returns [`Delta`][Temporality::Delta] for all
+/// instruments except `UpDownCounter` and `ObservableUpDownCounter`.
+///
+/// This temporality selector is equivalent to OTLP Metrics Exporter's
+/// `Delta` temporality preference (see [its documentation][exporter-docs]).
+///
+/// [exporter-docs]: https://github.com/open-telemetry/opentelemetry-specification/blob/a1c13d59bb7d0fb086df2b3e1eaec9df9efef6cc/specification/metrics/sdk_exporters/otlp.md#additional-configuration
+#[non_exhaustive]
+#[derive(Debug)]
+pub struct DeltaTemporalitySelector;
```

Comment:
on second thought, do we need this to be pub? I am thinking maybe this is better moved to the sdk itself as a pub struct.

The `pub fn with_delta_temporality` alone should be sufficient for users to leverage this, right?

Reply:
Ah well, if it's public it certainly shouldn't be `#[non_exhaustive]` (or it needs a c'tor). I'm not sure what the best solution is, I kind of dislike making it private since that makes it impossible to reuse its `TemporalitySelector` impl as part of another `TemporalitySelector` impl, but I have no idea how useful that would actually be.

Also seems like it might pollute the root namespace a little bit, but since the `metric` module is private, re-exporting at the crate root seemed like the only sensible choice.

Reply:
Changing temporality is a very common ask, so I am anticipating that OTel SDK itself will provide the building blocks for that, so that every exporter can leverage it!
There is a lot of duplication today for this!
eg: https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-sdk/src/metrics/mod.rs#L603

let me think more and come back. Hopefully there are others who can chime in with a better solution.

Reply:
Within core metrics Exporters, the OTLP exporter is the only valid scenario (per specs) for providing Delta Temporality configuration, as Prometheus is cumulative. If so, we can let it be here. Other vendor-specific exporters can independently decide on implementing their temporality selector, including the option for delta, if necessary.

Reply:
we can still achieve the goal without making the struct public, just the builder method public - @lalitb is this possible?

Reply:
Yes, that should be the right way.